syntax = "proto3";
// python -m grpc_tools.protoc -I./protos --python_out=. --grpc_python_out=. ./protos/spelling_bee_game.proto
package app;

service SpellingBeeGame{
    rpc CreateGame(GameRequest) returns (GameResponse) {}
    rpc FinalizeGame (FinalizeRequest) returns (FinalizeResponse) {}
    rpc ProcessSuggestion (SuggestionRequest) returns (SuggestionResponse) {}
}


message GameRequest {
    string userName = 1;
    string gameType = 2;
}

message GameResponse {
    bytes gameId = 1;
}

message FinalizeRequest {
    bytes gameId = 1;
}

message FinalizeResponse {
    string pangram = 1;
}

message Word {
    string word = 1;
}

message SuggestionRequest {
    bytes gameId = 1;
    int32 playerIndex = 2;
    Word suggestion = 3;
}

message SuggestionResponse {
    bool result = 1;
    string message = 2;
}